src/app/user-settings/user-settings-store.js

```
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 The "MysteriumNetwork/mysterion" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// @flow
+import {readFile, writeFile} from 'fs'
+import {promisify} from 'util'
+import type {UserSettings} from './user-settings'
+
+const readFileAsync = promisify(readFile)
+const writeFileAsync = promisify(writeFile)
+
+const defaultSettings: UserSettings = {
+  showDisconnectNotifications: true
+}
+
+async function saveSettings (path: string, settings: UserSettings): Promise<?Error> {
+  const settingsString = JSON.stringify(settings)
+  await writeFileAsync(path, settingsString)
+}
+
+async function loadSettings (path: string): Promise<?UserSettings> {
+  const data = await readFileAsync(path, {encoding: 'utf8'})
+  const parsedSettings = JSON.parse(data)
+
+  if (typeof parsedSettings.showDisconnectNotifications !== 'boolean') {
+    return null
```

We can just throw an error in this case - i.e. if JSON parsing fails in this method, we throw error, so this behaviour should be consistent.

---

done